app/services/foreman/provision/ssh.rb

```
@@ -99,10 +99,19 @@ def initiate_connection!
         retry
       rescue Timeout::Error
         retry
+      rescue IOError
+        logger.debug "net-ssh threw an IOError, retrying"
```

Please do not swallow exceptions, name the exception and properly log it.

    Foreman::Logging.exception("IO error occurred", exception)